scripts/functions.sh

```diff
@@ -341,24 +341,23 @@ function login_as_launchpad {
     echo ""
     echo "Getting launchpad coordinates from subscription: ${TF_VAR_tfstate_subscription_id}"
 
-    export keyvault=$(az keyvault list --subscription ${TF_VAR_tfstate_subscription_id} --query "[?tags.tfstate=='${TF_VAR_level}' && tags.environment=='${TF_VAR_environment}']" -o json | jq -r .[0].name)
-
-    echo " - keyvault_name: ${keyvault}"
+    keyvault=$(az graph query -q "Resources | where type == 'microsoft.keyvault/vaults' and tags.tfstate == '${TF_VAR_tfstate_subscription_id}' and  tags.environment == '${TF_VAR_environment}' and tags.tfstate == '${TF_VAR_level}'  | project name" | jq -r .data[0])
+    echo " - keyvault_name: $(echo "${keyvault}" | jq -r .name )"
 
     stg=$(az storage account show --ids ${id} -o json)
 
-    export TF_VAR_tenant_id=$(az keyvault secret show --subscription ${TF_VAR_tfstate_subscription_id} -n tenant-id --vault-name ${keyvault} -o json | jq -r .value) && echo " - tenant_id : ${TF_VAR_tenant_id}"
+    export TF_VAR_tenant_id=$(echo "${keyvault}" | jq -r .tenantId ) && echo " - tenant_id : ${TF_VAR_tenant_id}"
```

Comment:
Will the value of the secret that would be stored in the keyvault will always match the tenant id of the keyvault itself? In other words, are we sure this replacement is equivalent?

Reply:
Good catch, wrong assumption on my part. I thought it would save an unnecessary API call.